server/implementation/src/main/java/io/smallrye/graphql/execution/metrics/MetricsEmitter.java

```diff
@@ -0,0 +1,57 @@
+package io.smallrye.graphql.execution.metrics;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.jboss.logging.Logger;
+
+import io.smallrye.graphql.api.Context;
+import io.smallrye.graphql.config.ConfigKey;
+import io.smallrye.graphql.spi.MetricsService;
+import io.smallrye.graphql.spi.config.Config;
+
+public class MetricsEmitter {
+
+    private static final Logger LOG = Logger.getLogger(MetricsEmitter.class);
+    private final List<MetricsService> enabledServices;
+
+    public static MetricsEmitter getInstance() {
+        return new MetricsEmitter();
+    }
+
+    private MetricsEmitter() {
+        Config config = Config.get();
+        boolean enabled = config.getConfigValue(ConfigKey.ENABLE_METRICS, boolean.class, false);
+
+        // Find out which metrics API is available
+        ServiceLoader<MetricsService> metricService = ServiceLoader.load(MetricsService.class);
+        Iterator<MetricsService> it = metricService.iterator();
+        List<MetricsService> enabledServices = new ArrayList<>();
+
+        while (enabled && it.hasNext()) {
+            try {
+                enabledServices.add(it.next());
+            } catch (Throwable t) {
+                // Ignore that service...
+                Throwable cause = t.getCause();
+                LOG.debug("Failed to register " + t.getMessage()
```

Comment:
Can we remove this? Given that a `MetricsEmitter` is created multiple times (at least once for every operation) this can pollute the log unnecessarily

Reply:
Yes, of course. Should I also delete the same log which is in the `EventEmitter` class?

Reply:
I'd keep that one, because in this case it's not quite "expected" that the constructor throws an exception. In the metrics case, it will be quite usual and it won't necessarily be a sign of an actual error.